### ISSUE TITLE
Add a GitHub Action to run cspell.

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#5811]
This is a sibling task to others, like #5245, #5296, and #5467.

Relevant information:
- CSpell: https://cspell.org | https://github.com/streetsidesoftware/cspell
- GitHub Action: https://github.com/marketplace/actions/cspell-action (there may be others to choose from)
- Issue where CSpell was added to D9 core: [Add .cspell.json to automate spellchecking in Drupal core](https://www.drupal.org/project/drupal/issues/2972224)
- Issue to add cspell to the DrupalCI: [Support code spell checking](https://www.drupal.org/project/drupalci/issues/3122096)
- The dictionary currently in use at the time of writing: https://git.drupalcode.org/project/drupal/-/blob/9.5.x/core/misc/cspell/dictionary.txt
- [Rename dictionary.txt file to drupal.dic to make it compatible with PhpStorm on Linux](https://www.drupal.org/project/drupal/issues/3153919)

**Comment by BWPanda** [comment]
Here's a PR: https://github.com/backdrop/backdrop/pull/4356

It:
- Adds a custom dictionary for Backdrop (`.cspell/backdrop.dic`)
- Adds the CSpell config file (`.cspell/cspell.json`)
- Adds a new job to the existing PHP Lint action to check spelling of changes files in PRs

I think we may want to wait to merge this PR until the one from https://github.com/backdrop/backdrop-issues/issues/5987 gets in first. That way only typos made in PRs will be flagged (as opposed to existing typos in core).

**Comment by klonos** [comment]
Hey @BWPanda with #5987 now merged, can you please rebase your PR branch? We should add this in before we start introducing more typos and spelling mistakes.

**Comment by BWPanda** [comment]
@klonos Done.

**Comment by klonos** [comment]
@BWPanda nice 👍🏼 ...it already caught a typo:

```
Warning: Unknown word (Hepler)
core/modules/entityreference/tests/entityreference.form.test:181:6 Unknown word (Hepler)
```

Let's fix that then and merge this 😉

**Comment by BWPanda** [comment]
Typo fixed.

**Comment by BWPanda** [comment]
I don't particularly like adding specialized comments to code for ignoring coding standards, spellcheck, etc. It makes code harder to read/understand IMO.

I tried to ignore the very minimum of files. Is there anything in particular you're concerned about?

**Comment by quicksketch** [comment]
Note to self: We'll need to ignore the .cspell directory from the packager.

**Comment by BWPanda** [comment]
@klonos The `overrides` section ignores specific words in specific files. For example, in `core/modules/layout/tests/layout.test` the word "isnt" is ignored as a typo (i.e. it's allowed). All other typos are flagged in that file though, and the word "isnt" is still flagged in all other files.

The `ignorePaths` section, however, does ignore entire files. But I believe it's justified based on the ratio of 'typos' to actual, proper words in those files.

I personally don't prefer having a bunch of comments disabling and re-enabling the spellchecker throughout core, but if others agree that's better then so be it.

**Comment by klonos** [comment]
> All other typos are flagged in that file though, and the word "isnt" is still flagged in all other files.

Fair enough @BWPanda - I hadn't realized that.

@quicksketch mentioned during the weekly dev meeting that the preference would of course be to fix the underlying cause for the cases where for example `isnt` and `dont`need to exist in that form in the first place.

**Comment by quicksketch** [comment]
Considering this is only in a test, I think it would be fine to change the menu path and function to `do_not`.

**Comment by BWPanda** [comment]
Ok, PR updated. I renamed the path/function name as requested, and changed Black Widow's quote to one that doesn't use an apostrophe ;-)

**Comment by klonos** [comment]
Everything looks good to me now. ...I did leave some silly suggestions in the PR, but I don't want to block this. The longer we wait, the more the chances for some typo to creep into our codebase from another PR that gets merged before this.

**Comment by BWPanda** [comment]
> .I did leave some silly suggestions in the PR

Yeah, nah.

**Comment by klonos** [comment]
I'm adding the doco label, to remember to add this check in the"Automated tests" section of https://docs.backdropcms.org/documentation/how-to-use-the-core-issue-queue

(thanks @bugfolder for putting that together 🙏🏼 )

**Comment by klonos** [comment]
We should merge this in ASAP please.

**Comment by quicksketch** [comment]
Here's a PR to remove `.cspell` directories from the packager to avoid this directory ending up in our core releases: https://github.com/backdrop-ops/backdropcms.org/pull/985

**Comment by quicksketch** [comment]
I merged in https://github.com/backdrop/backdrop/pull/4356 and the update to BackdropCMS.org at https://github.com/backdrop-ops/backdropcms.org/pull/985. This is awesome! Thanks @BWPanda!!
